test/tar_interpreter_test.go

```diff
@@ -0,0 +1,156 @@
+package test
+
+import (
+	"archive/tar"
+	"bytes"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wal-g/wal-g/internal"
+)
+
+type testInterpretOptions struct {
+	DBDataDirectory string
+
+	Name     string
+	Typeflag byte
+
+	Create func(string) error
+	Delete func(string) error
+}
+
+func testInterpret(t *testing.T, options testInterpretOptions) (os.FileInfo, os.FileInfo) {
+	tarInterpreter := &internal.FileTarInterpreter{
+		DBDataDirectory: options.DBDataDirectory,
+	}
+
+	err := options.Create(options.Name)
+
+	assert.NoError(t, err)
+
+	defer options.Delete(options.Name)
+
+	err = tarInterpreter.Interpret(
+		&bytes.Buffer{},
+		&tar.Header{
+			Name:     options.Name,
+			Typeflag: options.Typeflag,
+		},
+	)
+
+	assert.NoError(t, err)
+
+	defer options.Delete(path.Join(options.DBDataDirectory, options.Name))
+
+	srcFileInfo, err := os.Lstat(options.Name)
+
+	assert.NoError(t, err)
+
+	dstFileInfo, err := os.Lstat(path.Join(options.DBDataDirectory, options.Name))
+
+	assert.NoError(t, err)
+
+	return srcFileInfo, dstFileInfo
+}
+
+func mkdirAll(path string) error {
+	return os.MkdirAll(path, 0766)
+}
+
+func justCreate(path string) error {
```

Comment:
Rename it to createFile and comment that it's needed for unification returning type with createDir.

Reply:
I think a comment is not needed, the purpose of the function becomes obvious if you look at the place of use.

Reply:
I think a comment is not needed, the purpose of the function becomes obvious if you look at the place of use.